src/hb-aat-layout-feat-table.hh

```diff
@@ -95,15 +94,19 @@ struct FeatureName
       len = MIN (settings_count - start_offset, *records_count);
       for (unsigned int i = 0; i < len; i++)
       {
-	records_buffer[i].is_default = exclusive && not_default &&
-				       i + start_offset == (featureFlags & IndexMask);
 	records_buffer[i].name_id = settings[start_offset + i].nameIndex;
 	records_buffer[i].setting = settings[start_offset + i].setting;
       }
-      if (exclusive && !not_default && start_offset == 0 && len != 0)
-        records_buffer[0].is_default = true;
     }
-    if (is_exclusive) *is_exclusive = exclusive;
+    if (default_setting)
+    {
+      if (exclusive)
+      {
+	if (settings_count && !not_default) *default_setting = settings[0].setting;
+	else if (not_default) *default_setting = featureFlags & IndexMask;
+      }
+      else *default_setting = HB_AAT_FEATURE_NO_DEFAULT_INDEX;
```

Comment:
This sounds like an index rather than a setting, too.

I do wonder if returning an index will be simpler? Otherwise, I'm not sure what the result for non-exclusive should be here.

Reply:
> This sounds like an index rather than a setting, too.

Right.

> I do wonder if returning an index will be simpler?

You are right. I was just thinking actual value may mean better on iteration (so it won't be unclear if index 3 is for this iteration or globally). Lets see what Behdad thinks.

> Otherwise, I'm not sure what the result for non-exclusive should be here.

Non-exclusive means not any selected also AFAICS.